terraform/eval_state.go

```diff
@@ -58,6 +57,25 @@ func (n *EvalReadState) Eval(ctx EvalContext) (interface{}, error) {
 	return result, nil
 }
 
+// EvalRequireState is an EvalNode implementation that early exits
+// if the state doesn't have an ID.
+type EvalRequireState struct {
+	State **InstanceState
+}
+
+func (n *EvalRequireState) Eval(ctx EvalContext) (interface{}, error) {
+	if n.State == nil {
+		return nil, EvalEarlyExitError{}
```

Comment:
Note: `EvalEarlyExitError` is a special error that is like `return` in code... just early exits and doesn't make an actual error.